tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java

```diff
@@ -518,19 +518,34 @@ public void workerVariableVisibilityTest() throws BallerinaTestException {
         debugTestRunner.addBreakPoint(new BallerinaTestDebugPoint(debugTestRunner.testEntryFilePath, 31));
         debugTestRunner.initDebugSession(DebugUtils.DebuggeeExecutionKind.RUN);
         Pair<BallerinaTestDebugPoint, StoppedEventArguments> debugHitInfo = debugTestRunner.waitForDebugHit(25000);
-
-        // variable visibility test for workers outside fork (workers are visible outside fork() as futures).
-        localVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), DebugTestRunner.VariableScope.LOCAL);
-        debugTestRunner.assertVariable(localVariables, "w1", "future<()>", "future");
-
-        // variable visibility test inside worker (only worker's variables should be visible).
-        debugTestRunner.resumeProgram(debugHitInfo.getRight(), DebugTestRunner.DebugResumeKind.NEXT_BREAKPOINT);
-        debugHitInfo = debugTestRunner.waitForDebugHit(10000);
-        localVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), DebugTestRunner.VariableScope.LOCAL);
-        debugTestRunner.assertVariable(localVariables, "x", "10", "int");
-
-        // variables outside worker should not be visible
-        Assert.assertFalse(localVariables.containsKey("a"));
+        if (debugHitInfo.getKey().getLine() == 23) {
```

Comment:
This fix is added because with new virtual thread implementation, the order of execution for workers is not consistent. Therefore, we can't rely on the debug point hitting order. I modified the assertions depending on which debug point hits first. 
@warunalakshitha Is this expected?

Reply:
Yes. That should be ok